src/pythia/src/utils/macros.rs

```diff
@@ -0,0 +1,46 @@
+#[macro_export]
+macro_rules! clone_with_state {
+    ($field:ident) => {{
+        $crate::STATE.with(|state| state.borrow().$field.clone())
+    }};
+}
+
+#[macro_export]
+macro_rules! update_state {
+    ($field:ident, $value:expr) => {{
+        $crate::STATE.with(|state| {
+            state.borrow_mut().$field = $value;
+        })
+    }};
+}
+
+#[macro_export]
+macro_rules! log {
+    ($($arg:tt)*) => {{
+        ic_cdk::println!($($arg)*);
+        ic_utils::logger::log_message(format!($($arg)*));
+        ic_utils::monitor::collect_metrics();
+    }};
+}
+
+#[macro_export]
+macro_rules! retry_until_success {
```

Comment:
I'm not sure about this macro, as it's each invocation will increase the bytecode size of wasm binary. I would just create a function for that